OpenRA.Mods.Common/HitShapes/Circle.cs

```diff
@@ -48,25 +51,27 @@ public WDist DistanceFromEdge(WVec v)
 
 		public WDist DistanceFromEdge(WPos pos, Actor actor)
 		{
-			var actorPos = actor.CenterPosition;
+			var orientation = actor.Orientation;
+			var actorPos = actor.CenterPosition + LocalOffset.Rotate(orientation);
 
 			if (pos.Z > actorPos.Z + VerticalTopOffset)
-				return DistanceFromEdge(pos - (actorPos + new WVec(0, 0, VerticalTopOffset)));
+				return DistanceFromEdge((pos - (actorPos + new WVec(0, 0, VerticalTopOffset))).Rotate(-orientation));
```

Comment:
The three points of interest (center of actor, center of actor + `v`, center of actor + `LocalOffset`) define a triangle, so you can use the [cosine rule](https://en.wikipedia.org/wiki/Law_of_cosines) to simplify the distance calculation in the other `DistanceFromEdge` overload.